ui/src/server_axum.rs

```diff
@@ -386,6 +391,19 @@ async fn metrics(_: MetricsAuthorization) -> Result<Vec<u8>, StatusCode> {
         .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)
 }
 
+async fn websocket(ws: WebSocketUpgrade) -> impl IntoResponse {
+    ws.on_upgrade(handle_socket)
+}
+
+async fn handle_socket(mut socket: WebSocket) {
+    LIVE_WS.inc();
+    let start = Instant::now();
+    while let Some(Ok(_msg)) = socket.recv().await {}
```

Comment:
What will be sending these messages? Does the JS side implicitly send keepalives? Or are we basically just waiting for the channel to shutdown and there won't be any messages sent here?

Reply:
For now, there is no message.
Keepalives are automatically handled on both sides.